src/riak_core_util.erl

```diff
@@ -704,6 +712,141 @@ proxy(Parent, Fun) ->
+%% You are STRONGLY advised to use job_class_enabled/2.
+job_class_enabled(Class) ->
+    case app_helper:get_env(riak_core, job_accept_class) of
+        undefined ->
+            true;
+        [] ->
+            false;
+        [_|_] = EnabledClasses ->
+            lists:member(Class, EnabledClasses);
+        Other ->
+            % Don't crash if it's not a list - that should never be the case,
+            % but since the value *can* be manipulated externally be more
+            % accommodating. If someone mucks it up, nothing's going to be
+            % allowed, but give them a chance to catch on instead of crashing.
+            _ = lager:error(
+                "riak_core.job_accept_class is not a list: ~p", [Other]),
+            false
+    end.
+
+-spec job_class_disabled_message(ReturnType :: atom(), Class :: term())
+        -> binary() | string().
+%% @doc The error message to be returned to a client for a disabled job class.
+%% WARNING:
+%% * This function is likely to be extended to accept a Job as well as a Class
+%%   when the Jobs API is live.
+job_class_disabled_message(binary, Class) ->
+    erlang:list_to_binary(job_class_disabled_message(text, Class));
+job_class_disabled_message(text, Class) ->
+    lists:flatten(io_lib:format("Operation '~p' is not enabled", [Class])).
+
+-spec report_job_request_disposition(Accepted :: boolean(), Class :: term(),
+    Mod :: module(), Func :: atom(), Line :: pos_integer(), Client :: term())
+        -> ok | {error, term()}.
+%% @doc Report/record the disposition of an async job request.
+%%
+%% Logs an appropriate message and reports to whoever needs to know.
+%% WARNING:
+%% * This function is likely to be extended to accept a Job as well as a Class
+%%   when the Jobs API is live.
+%%
+%% Parameters:
+%%  * Accepted - Whether the specified job Class is enabled.
+%%  * Class - The Class of the job, by convention {Application, Operation}.
+%%  * Mod/Func/Line - The Module, function, and source line number,
+%%    respectively, that will be reported as the source of the call.
+%%  * Client - Any term indicating the originator of the request.
+%%    By convention, when meaningful client identification information is not
+%%    available, Client is an atom representing the protocol through which the
+%%    request was received.
+%%
+report_job_request_disposition(true, Class, Mod, Func, Line, Client) ->
+    lager:log(info,
+        [{pid, erlang:self()}, {module, Mod}, {function, Func}, {line, Line}],
+        "Request '~p' accepted from ~p", [Class, Client]);
+report_job_request_disposition(false, Class, Mod, Func, Line, Client) ->
+    lager:log(warning,
+        [{pid, erlang:self()}, {module, Mod}, {function, Func}, {line, Line}],
+        "Request '~p' disabled from ~p", [Class, Client]).
+
 %% ===================================================================
 %% EUnit tests
 %% ===================================================================
```

Comment:
Would be good to have some simple unit tests around the enable/disable/enabled/disabled functions.